### ISSUE TITLE
MAC Pro M1:  '_FakeLoader' object has no attribute '_get_shared_seed'

### ISSUE THREAD

**Issue by geg00** [fastai/fastai#3704]
Be sure you've searched [the forums](https://forums.fast.ai) for the error message you received. Also, unless you're an experienced fastai developer, first ask on the forums to see if someone else has seen a similar issue already and knows how to solve it. **Only file a bug report here when you're quite confident it's not an issue with your local setup.**

**Please see [this model example](https://github.com/fastai/fastai2/issues/487) of how to fill out an issue correctly.  Please try to emulate that example as appropriate when opening an issue.**

Please confirm you have the latest versions of fastai, fastcore, and nbdev prior to reporting a bug (delete one): YES 

**Describe the bug**
After you define the DataBlock 
if you attempt to execute dls.show_batch() it fails.

**To Reproduce**

I'm using this Kaggle competition
https://www.kaggle.com/competitions/humpback-whale-identification
Use a Mac with M1
Install the latest fastai code (this one will install 1.11 (last stable code)
Install the nightly version of pytorch that supports MPS 1.12 or after

```
from fastcore.all import *
from fastai.vision.all import *
import time
import timm
import sys

df = pd.read_csv('./train.csv')
path = Path('.')

def get_x(r): return path/'train'/r['Image']
def get_y(r): return r['Id']

dls = DataBlock(
    blocks=(ImageBlock, CategoryBlock),
    splitter=RandomSplitter(valid_pct = 0.2, seed=42),
    get_x=get_x,
    get_y=get_y,
    item_tfms=Resize((256,192), method=ResizeMethod.Pad, pad_mode=PadMode.Zeros),
    batch_tfms=aug_transforms(size=(171,128), min_scale=0.75)
).dataloaders(df)

dls.show_batch()
```

**Expected behavior**
to show 6 samples of the dataset with their names

**Error with full stack trace**

Place between these lines with triple backticks:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/Users/gerardogarcia/whales/Untitled-1.ipynb Cell 7' in <cell line: 1>()
----> 1 dls.show_batch()

File ~/mambaforge/lib/python3.9/site-packages/fastai/data/core.py:152, in TfmdDL.show_batch(self, b, max_n, ctxs, show, unique, **kwargs)
    150     old_get_idxs = self.get_idxs
    151     self.get_idxs = lambda: Inf.zeros
--> 152 if b is None: b = self.one_batch()
    153 if not show: return self._pre_show_batch(b, max_n=max_n)
    154 show_batch(*self._pre_show_batch(b, max_n=max_n), ctxs=ctxs, max_n=max_n, **kwargs)

File ~/mambaforge/lib/python3.9/site-packages/fastai/data/load.py:177, in DataLoader.one_batch(self)
    175 def one_batch(self):
    176     if self.n is not None and len(self)==0: raise ValueError(f'This DataLoader does not contain any batches')
--> 177     with self.fake_l.no_multiproc(): res = first(self)
    178     if hasattr(self, 'it'): delattr(self, 'it')
    179     return res

File ~/mambaforge/lib/python3.9/site-packages/fastcore/basics.py:626, in first(x, f, negate, **kwargs)
    624 x = iter(x)
    625 if f: x = filter_ex(x, f=f, negate=negate, gen=True, **kwargs)
--> 626 return next(x, None)

File ~/mambaforge/lib/python3.9/site-packages/fastai/data/load.py:132, in DataLoader.__iter__(self)
...
--> 587     self._shared_seed = loader._get_shared_seed()
    588     if isinstance(self._dataset, IterDataPipe):
    589         shared_rng = torch.Generator()

AttributeError: '_FakeLoader' object has no attribute '_get_shared_seed'
```

**Additional context**

Hardware Overview:

  Model Name:	MacBook Pro
  Model Identifier:	MacBookPro18,2
  Chip:	Apple M1 Max
  Total Number of Cores:	10 (8 performance and 2 efficiency)
  Memory:	32 GB
  System Firmware Version:	7459.121.3
  OS Loader Version:	7459.121.3
  Activation Lock Status:	Enabled

**Comment by muellerzr** [comment]
@geg00 what version of fastai do you have? I can't find that dataloader code anywhere.

Please do `pip show fastai` and report that version.

**Comment by geg00** [comment]
(base) gerardogarcia@Gerardos-MacBook-Pro ~ % pip show fastai
Name: fastai
Version: 2.7.3
Summary: fastai simplifies training fast and accurate neural nets using modern best practices
Home-page: https://github.com/fastai/fastai/tree/master/
Author: Jeremy Howard, Sylvain Gugger, and contributors
Author-email: info@fast.ai
License: Apache Software License 2.0
Location: /Users/gerardogarcia/mambaforge/lib/python3.9/site-packages
Requires: fastcore, fastdownload, fastprogress, matplotlib, packaging, pandas, pillow, pip, pyyaml, requests, scikit-learn, scipy, spacy, torch, torchvision
Required-by: 
(base) gerardogarcia@Gerardos-MacBook-Pro ~ % pip show fastcore
Name: fastcore
Version: 1.4.5
Summary: Python supercharged for fastai development
Home-page: https://github.com/fastai/fastcore/tree/master/
Author: Jeremy Howard and Sylvain Gugger
Author-email: infos@fast.ai
License: Apache Software License 2.0
Location: /Users/gerardogarcia/mambaforge/lib/python3.9/site-packages
Requires: packaging, pip
Required-by: fastai, fastdownload
(base) gerardogarcia@Gerardos-MacBook-Pro ~ %

**Comment by geg00** [comment]
https://github.com/fastai/fastai/issues/3655

**Comment by jph00** [comment]
Sorry we don't support Macs.

**Comment by geg00** [comment]
😐

**Comment by jim-king-2000** [comment]
Met the same issue.
https://github.com/fastai/fastai/issues/3655#issuecomment-1170663578

### Software env:
| name | version |
| ------ | -------- |
| Ubuntu | 22.04 |
| Linux | Linux 5.15.0-40-generic |
| cuda | 11.7 |
| torch | 1.12.0+cu116 |
| torchaudio | 0.12.0+cu116 |
| torchvision | 0.13.0+cu116 |
| fastai | 2.7.4 |

### Hardware env:
| name | version |
| ------ | -------- |
| CPU | 12th Gen Intel(R) Core(TM) i5-12400 |
| memory | 16GB |
| GPU |  NVIDIA Corporation GP108 [GeForce GT 1030] (rev a1) 2GB |

**Comment by jim-king-2000** [comment]
It's been fixed on v2.7.5. But there are still some warnings:

```
/home/jim/.local/lib/python3.10/site-packages/torchvision/models/_utils.py:135: UserWarning: Using 'weights' as positional parameter(s) is deprecated since 0.13 and will be removed in 0.15. Please use keyword parameter(s) instead.
  warnings.warn(
/home/jim/.local/lib/python3.10/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and will be removed in 0.15. The current behavior is equivalent to passing `weights=ResNet34_Weights.IMAGENET1K_V1`. You can also use `weights=ResNet34_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
```
